### ISSUE TITLE
Unable to load/save session on macOS

### ISSUE THREAD

**Issue by rafi** [rmagatti/auto-session#34]
Using macOS, unable to load session on startup, using this configuration:
```lua
local opts = {
	log_level = 'debug',
	auto_session_enable_last_session = false,
	auto_session_root_dir = vim.fn.stdpath('data')..'/sessions/',
	auto_session_enabled = true,
	auto_save_enabled = true,
	auto_restore_enabled = true,
	auto_session_suppress_dirs = {'/etc', vim.fn.expand('$HOME')}
}

require('auto-session').setup(opts)
```

It starts working once I patch `is_readable` like this:
```diff
diff --git i/lua/auto-session-library.lua w/lua/auto-session-library.lua
index c879f650..3d4cfdf8 100644
--- i/lua/auto-session-library.lua
+++ w/lua/auto-session-library.lua
@@ -144,11 +144,8 @@ function Lib.legacy_session_name_from_cwd()
 end

 function Lib.is_readable(file_path)
-  if IS_WIN32 then
-    file_path = file_path:gsub('\\%%','%%')
-  end
-
-  return vim.fn.filereadable(vim.fn.expand(file_path)) ~= Lib._VIM_FALSE
+  file_path = file_path:gsub('\\%%','%%')
+  return vim.fn.filereadable(file_path) ~= Lib._VIM_FALSE
 end
 -- ===================================================================================
```

**Comment by rmagatti** [comment]
Hm this is odd, I'm also on MacOS and loading sessions has been working fine. Does that patch work because it's for some reason going into the `IS_WIN32` conditional? 🤔

**Comment by rafi** [comment]
If you notice the last two lines I left, I must substitute `\%` => `%`, and cancel `vim.fn.expand()` for it to work.
I find it weird that it works for you, session files are written with `%` instead of `/`, without backslashes...
And `expand()` returns an empty string, not sure why you use expand...

I'm on latest HEAD of NVIM v0.5.0-dev+93f15db5d

**Comment by rmagatti** [comment]
Can you paste me your result of `:messages` so I can check it out since you have log level set to debug?

I'm not on latest master yet, I'll upgrade and try it out.

**Comment by rmagatti** [comment]
So the `file_path` that gets to that function looks like this
`"/Users/rmagatti/.local/share/nvim/sessions/\\%Users\\%rmagatti\\%Projects\\%auto-session.vim"`.

If you try and do a readable on that raw path, it returns `0`.

Now if you do `vim.fn.expand(file_path)` this is the result `/Users/ronnieandrewmagatti/.local/share/nvim/sessions/%Users%ronnieandrewmagatti%Projects%auto-session.vim`, which is identical to the `gsub` result.

So basically, both cases should be working fine for you. Either way I can change this to just do gsub instead since it's identical.

**Comment by rafi** [comment]
Works great, thanks.
